### PR TITLE
coz: init at 0.2.0

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

profiling nix

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by tobim** [comment]
I would suggest to build the viewer as well (https://github.com/plasma-umass/coz/tree/master/viewer).

**Review by wykurz — APPROVED** [review]

File: pkgs/development/tools/analysis/coz/default.nix
Comment (tobim) [inline pkgs/development/tools/analysis/coz/default.nix]: `node` is only required for the viewer, which is not built from this expression.

File: pkgs/development/tools/analysis/coz/default.nix
Comment (tobim) [inline pkgs/development/tools/analysis/coz/default.nix]: `ncurses` is only used for logging in the `Makefile`s, so it should be moved to `nativeBuildInputs`.

File: pkgs/development/tools/analysis/coz/default.nix
Comment (tobim) [inline pkgs/development/tools/analysis/coz/default.nix]: These 2 don't show up in `nix-store -q --references `, they should be moved to `propagatedBuildInputs`.

**Review by tobim — CHANGES_REQUESTED** [review]
nit: For some reason `include/coz.h` is marked as executable in the store.

Maybe the accompanying viewer should be built as well, I don't know how useful this is without it.

**Comment by zimbatm** [comment]
Thanks @tobim for your review. I need to use coz a bit before this is ready to be merged.

coz is currently not able to find libcoz.so. This is already fixed in master upstream so I'll just wait for a new release.

File: pkgs/development/tools/analysis/coz/default.nix
Comment (zimbatm) [reply pkgs/development/tools/analysis/coz/default.nix]: fixed in 5a1aef8468685e3586be5b8e53cbdb91ade7fe64

File: pkgs/development/tools/analysis/coz/default.nix
Comment (zimbatm) [reply pkgs/development/tools/analysis/coz/default.nix]: fixed in 5a1aef8468685e3586be5b8e53cbdb91ade7fe64

File: pkgs/development/tools/analysis/coz/default.nix
Comment (zimbatm) [reply pkgs/development/tools/analysis/coz/default.nix]: fixed in 5a1aef8468685e3586be5b8e53cbdb91ade7fe64. I'll do the frontend later

**Comment by zimbatm** [comment]
Currently hitting this issue where the program fails if PYTHONPATH is set. The script should probably be more hermetic to the environment:
```
$ /nix/store/y81ql1b3vgw6gl13zrsm91x72wqnh8bm-coz-0.2.0/bin/coz run --- ./inst/bin/nix-env -q
Fatal Python error: initsite: Failed to import the site module
Traceback (most recent call last):
  File "/nix/store/vvr1zh2v02dw3i4wd2gq3xnv0l3cjlv5-python2.7-setuptools-40.8.0/lib/python2.7/site-packages/site.py", line 73, in <module>
    __boot()
  File "/nix/store/vvr1zh2v02dw3i4wd2gq3xnv0l3cjlv5-python2.7-setuptools-40.8.0/lib/python2.7/site-packages/site.py", line 26, in __boot
    import imp  # Avoid import loop in Python 3
  File "/nix/store/k0igqpz995nmxwf6piyd5jjpcc7gcnyc-python3-3.7.4/lib/python3.7/imp.py", line 27, in <module>
    import tokenize
  File "/nix/store/k0igqpz995nmxwf6piyd5jjpcc7gcnyc-python3-3.7.4/lib/python3.7/tokenize.py", line 33, in <module>
    import re
  File "/nix/store/k0igqpz995nmxwf6piyd5jjpcc7gcnyc-python3-3.7.4/lib/python3.7/re.py", line 143, in <module>
    class RegexFlag(enum.IntFlag):
AttributeError: module 'enum' has no attribute 'IntFlag'
```
I'm not sure why there is a python2.7 path in there or why enu

**Pull request merged**
